src/lib/registration/ui/addon_eula_dialog.rb

```diff
@@ -116,23 +112,35 @@ def setup_eula_dialog(addon, eula_reader, tmpdir)
       # @return [Symbol] :accepted, :back, :abort, :halt - user input
       def run_eula_dialog(eula_reader)
         base_product = false
-        cancel_action = "abort"
+        cancel_action = "refuse"
         ret = Yast::ProductLicense.HandleLicenseDialogRet(arg_ref(eula_reader.licenses),
           base_product, cancel_action)
         log.debug "EULA dialog result: #{ret}"
+
         ret
       end
 
       # ask user to accept an addon EULA
       # @param [Addon] addon the addon
-      # @return [Symbol] :accepted, :back, :abort, :halt
+      # @return [Symbol] :next, :back, :abort, :halt
       def accept_eula(addon)
         Dir.mktmpdir("extension-eula-") do |tmpdir|
           return :back unless download_eula(addon, tmpdir)
-          eula_reader = EulaReader.new(tmpdir)
 
+          eula_reader = EulaReader.new(tmpdir)
           setup_eula_dialog(addon, eula_reader, tmpdir)
-          run_eula_dialog(eula_reader)
+          result = run_eula_dialog(eula_reader)
+
+          case result
+          when :accepted
+            addon.accept_eula
+            :next
+          when :refuse
```

Comment:
Um, the results should be symmetric, if "accept**ed**" already exists it should be "refuse**d**".

Reply:
You're right! I'll change it.